include/arkode/arkode_lsrkstep.h

```diff
@@ -34,11 +34,11 @@ typedef int (*ARKDomEigFn)(sunrealtype t, N_Vector y, N_Vector fn,
 
 typedef enum
 {
-  ARKODE_LSRK_RKC_2    = 1, /* ensure enum is int */
-  ARKODE_LSRK_RKL_2    = 2,
-  ARKODE_LSRK_SSP_S_2  = 4,
-  ARKODE_LSRK_SSP_S_3  = 5,
-  ARKODE_LSRK_SSP_10_4 = 6
+  ARKODE_LSRK_RKC_2    = 0, /* ensure enum is int */
+  ARKODE_LSRK_RKL_2    = 1,
```

Comment:
The major benefit to having a known value is that users can specify an integer value in their own programs, and this can be automatically converted to the correct `enum` value when passing the value to SUNDIALS.  Every time we take what was a known integer for a method and remove its value, consuming packages must then jump through hoops to allow control over our inputs from their own input files or command-line arguments.  While it's great that `enum` values help us to ensure legal inputs via compiler type-checking, I see absolutely no reason to hide the values from users.  As a case in point, our current "logging" documentation describes log "levels" via opaque names `SUN_LOGLEVEL_ALL`, `SUN_LOGLEVEL_NONE`, etc.  Then our own CMake system requires users to specify the logging level via an *integer*, even though the values (e.g., `SUN_LOGLEVEL_INFO=3`) are completely missing from our documentation. 

That said, I don't think that the values necessarily need to be grouped.  While we will definitely add more RKC/RKL methods (including ImEx and split-explicit versions) in the near future, those could be appended to the end of the current list.

Reply:
The point of having an `enum` is that users should not use specific (magic) integer values and instead use the enumerators.

Reply:
This makes perfect sense for a user who plans to hard-code their method type, in which case I agree that the enumerator is more descriptive than a "magic" integer.  

However, we are in the business of providing a useful tool for our users, many of whom build a software infrastructure for their own users, and who would like to control our inputs from their own configuration infrastructure.  Here, I imagine a F2003 code that uses namelists for inputs, or a C code that gets command-line inputs.  If we hide the values of our enums, then all of those users must set up `if/else` statements to run through each of our enumeration values.  Even in the cases where we create SUNDIALS utility routines to convert from strings to enumeration values, then the consuming package is still required to do string manipulation from their own inputs to ours.

However, if we just leave the enumeration values intact, and then keep ourselves from changing them frequently, all of the above steps can be simplified dramatically.

I personally see absolutely no reason to hide information and complicate the jobs of our users, particularly in the case of enumerations with values that we will retain for the long term.

Reply:
I'll echo David in that I think users should not use specific integer values directly. Similar to Butcher tables, we can have a function to map a string name to the ID to help with command line parsing. We can also add min and max values of the enum to support the case of looping over all possible LSRK methods.

Reply:
The real solution here is for us to have a way to parse command line or input file options into sundials options. Since we do not have that, the burden falls on the user codes to convert inputs to sundials options. With the number of options we have, that is a big ask and something we should address in the near future and make a high priority. 

However, I do not think the right choice is to expose the integer values so that user codes can "pass-through" the integer values. 1) It makes it difficult for us to ever change these values 2) it result in input files in the user codes that can easily be misleading, e.g., a user-code developer accidentally leaves a comment that " 1 = <some method>" when in reality it "1 = <some other method>" (I have seen this happen). 

For what its worth, if you don't specify the integer values in the enum, they are still predictably defined so a user could deduce the integer values and use them (but it will be obvious we are not condoning it).